### PR TITLE
Add GitHub Action workflow for tests

### PR DESCRIPTION
**Pull request by suz-stripe**
### Reviewers
r? @tomer-stripe 
cc @stripe/dev-platform

 ### Summary
+ Added a GitHub Action Workflow to run tests on Linux, Windows, and MacOS.
+ Fixed some issues in tests that failed within a Windows environment
+ Cleaned up Makefile to remove redundant separate install of `misspell` as it was failing to install on Windows. Looking at the `golinter-ci` config it appears that misspell is already installed and runs in this task with the other linters. Is this right?

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -34,18 +34,14 @@ fmt:
 .PHONY: fmt
 
 # Run all the linters
-lint: bin/golangci-lint bin/misspell
```
Comment (tomer-stripe) [inline Makefile:37]: Do we need to remove `misspell`?

Comment (suz-stripe) [reply Makefile:37]: My third bullet in the PR summary briefly explains this. I thought misspell is [already installed](https://golangci-lint.run/usage/linters) and used by out `golangci-lint' step by enabling it in the [golangci.yml file](https://github.com/stripe/stripe-cli/blob/master/.golangci.yml#L23)?

Comment (suz-stripe) [reply Makefile:37]: following up - I was able to provoke mispell to surface an error locally by running `make lint` even with the mispell install step removed.

File: pkg/cmd/root_test.go [hunk 2]
```diff
@@ -27,15 +28,15 @@ func executeCommandC(root *cobra.Command, args ...string) (c *cobra.Command, out
 func TestGetPathNoXDG(t *testing.T) {
 	actual := Config.GetConfigFolder("")
 	expected, err := homedir.Dir()
-	expected += "/.config/stripe"
```
Comment (swornov) [inline pkg/cmd/root_test.go:30]: expected += "/.config/stripe"

File: pkg/cmd/root_test.go [hunk 3]
```diff
@@ -27,15 +28,15 @@ func executeCommandC(root *cobra.Command, args ...string) (c *cobra.Command, out
...
+	expected += filepath.Join("/", ".config", "stripe")
 
 	require.NoError(t, err)
 	require.Equal(t, actual, expected)
 }
 
 func TestGetPathXDG(t *testing.T) {
 	actual := Config.GetConfigFolder("/some/xdg/path")
-	expected := "/some/xdg/path/stripe"
```
Comment (swornov) [inline pkg/cmd/root_test.go:38]: expected := "/some/xdg/path/stripe"

File: pkg/login/client_login_test.go [hunk 4]
```diff
@@ -119,13 +120,21 @@ func TestGetLinksHTTPStatusError(t *testing.T) {
...
+		errorString = "connect: connection refused"
+	}
+
 	// Immediately close the HTTP server so that the request fails.
 	ts := httptest.NewServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {}))
 	ts.Close()
 
 	links, err := getLinks(ts.URL, "test")
 	require.Error(t, err)
-	require.Contains(t, err.Error(), "connect: connection refused")
```
Comment (swornov) [inline pkg/login/client_login_test.go:128]: require.Contains(t, err.Error(), "connect: connection refused")

File: pkg/login/poll_test.go [hunk 5]
```diff
@@ -120,13 +121,21 @@ func TestHTTPStatusError(t *testing.T) {
...
+		errorString = "connect: connection refused"
+	}
+
 	// Immediately close the HTTP server so that the poll request fails.
 	ts := httptest.NewServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {}))
 	ts.Close()
 
 	response, account, err := PollForKey(ts.URL, 1*time.Millisecond, 3)
 	require.Error(t, err)
-	require.Contains(t, err.Error(), "connect: connection refused")
```
Comment (swornov) [inline pkg/login/poll_test.go:129]: require.Contains(t, err.Error(), "connect: connection refused")

**Pull request merged**